docs/element-definitions.md

```diff
@@ -1,1419 +1,1612 @@
-| JSON primitive |      Refract     | [MSON][] Base Type | Data Structure Elements |
-|:--------------:|:----------------:|:------------------:|:------------------------:|
-|     boolean    |  [Boolean Element][] |     boolean    |  Boolean Type  |
-|     string     |  [String Element][]  |     string     |   String Type  |
-|     number     |  [Number Element][]  |     number     |   Number Type  |
-|      array     |   Array Element  |      array     |   Array Type   |
-|        -       |         -        |      enum      |    Enum Type   |
-|     object     |  Object Element  |     object     |   Object Type  |
-|      null      |   Null Element   |        -       |        -       |
-
-### Data Structure Element (Base API Element)
+- `element`: link (string, fixed)
+- `attributes`
+    - `relation` (string) - Link relation type as specified in [RFC 5988](https://tools.ietf.org/html/rfc5988).
+    - `href` (string) - The URI for the given link
+    - `validation` - _reserved for future use_
 
-Base element for every Data Structure element.
+### Example
 
-The Data Structure Element adds attributes representing Data Structure _Type Definition_ and _Type Sections_.
+The following shows a link with the relation of `foo` and the URL of `/bar`.
 
-Note: In Data Structure Refract _Nested Member Types_ _Type Section_ is the `content` of the element.
+```json
+{
+  "element": "link",
+  "attributes": {
+    "relation": "foo",
+    "href": "/bar"
+  }
+}
+```
 
-#### Properties
+## Profiles
 
-- `attributes`
-    - `typeAttributes` (Array Element) - _Type Definition_ attributes list, see _Type Attribute_  
+The primary means by which users can provide semantic definitions and other meta information is through a profile. A profile MAY provide semantic information about an element and its data, it MAY provide specific instructions about elements such as how inheritance should work or how elements should be processed, and it MAY be used to modify understanding of existing elements in other profiles. The usage of a profile is not limited to these uses here, and SHOULD be left up to the profile author to define its use.
 
-        - `content` (array, fixed-type)
+To point to a profile, you MAY use the [profile link relation](https://www.ietf.org/rfc/rfc6906.txt) as a meta link in your root element or in any other element. Profile links may also be found outside of the document itself in places like the [HTTP Link Header](http://www.w3.org/wiki/LinkHeader). Additionally, a profile link is not required in order to use the functionality a profile provides, as a media type MAY define the same things a profile.
 
-            Type attributes of a type definition.
+Below is an example of how a profile link is used as a meta link.
 
-            Note, if `sample` (or `default`) attribute is specified the value SHOULD be stored in the `samples` (or `default`) property instead of the element's `content`.
+```json
+{
+  "element": "foo",
+  "meta": {
+    "links": [
```

Comment:
This isn't valid JSON serialisation, the links should be an array element, meta values MUST be elements themselves.

```json
{
    "element": "array",
    "content": [
        { "element": "link", ... }
    ]
}
```